src/hooks/useMutation/useMutation.ts

```diff
@@ -17,18 +14,15 @@ import { DataForMutation, GetMutationParams } from './useMutation.types';
  * @see ApiClientContextController.ts
  * */
 
-export const useMutation = <Key extends keyof AxiosMutationsType, TError = unknown>(
+export const useMutation = <Key extends keyof AxiosMutationsType, TError = ApiError>(
   mutation: Key,
-  options?: UseMutationOptions<DataForMutation<Key>, TError>,
+  options?: UseMutationOptions<DataForMutation<Key>, TError, GetMutationParams<Key>> & {
+    meta?: Partial<ExtendedQueryMeta>;
+  },
 ) => {
   const { client } = useApiClient();
   const mutationFn = mutations[mutation](client);
   const mutationKey: MutationKey = [mutation];
 
-  return useRQMutation(
-    mutationKey,
-    async (args) => await mutationFn(args),
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    options as any,
-  ) as UseMutationResult<DataForMutation<Key>, TError, GetMutationParams<Key>>;
```

Comment:
This typecasting was used to properly infer error type. Since now options are not typecasted to `any` Error type is properly inferred from `options` type